### ISSUE TITLE
ragged 2d kde boundaries

### ISSUE THREAD

**Issue by lukashergt** [williamjameshandley/anesthetic#94]
The contours of 2d `'kde'` plots are ragged when they reach up to the boudary as can be seen in the `[logr][n_s]` plot in the MWE below. It would be nicer (and more correct strictly speaking) for them to reach up to the boundary.

If I remember correctly this was introduced with #73, which aimed at handling highly correlated samples (such as in the `[logr][r]` plot below) better. #73 already states that it's only a partial fix and should be improved upon in the future. Hence, I thought it might be good to keep this issue in mind in future improvement attempts (e.g. when working on #69).

### MWE:

data: [unilog_r.log](https://github.com/williamjameshandley/anesthetic/files/4746305/unilog_r.log) (same data as in #93)

```python
data = pandas.read_csv("unilog_r.log", sep='\t')
ns = NestedSamples(data=data[['n_s', 'r', 'logr']], logL=data.logL, logL_birth=data.logL_birth, w=data.weight)
fig = plt.figure(figsize=(10, 10))
fig, axes = make_2d_axes(['n_s', 'r', 'logr'], fig=fig)
ns.plot_2d(axes);
```

![unilog_triangle](https://user-images.githubusercontent.com/18258042/84037600-b9771880-a996-11ea-81ad-81681491ffec.png)

**Comment by lukashergt** [comment]
For anybody with a particularly ugly case, one way to improve on this issue is by cranking up the value for `ncompress` which by default is set to 1000. In the OP example the last line should hence be changed, e.g. to:
```python
ns.plot_2d(axes, ncompress=5000);
```

**Comment by williamjameshandley** [comment]
For reference, could you provide an example in this discussion? 

I would leave the issue open, since hopefully future work on alternative kde schemes (e.g. #131) may resolve this.

**Comment by williamjameshandley** [comment]
This is partially resolved by cranking up ncompress and decreasing the bandwidth #193. However, this could also likely be resolved by adjusting the grid of points to include a set of additional _plotting points_ (rather than kde points) on the boundary.

**Comment by lukashergt** [comment]
Fixed by #216.
